### ISSUE TITLE
PartialType and IntersectionType doesn't work as expected

### ISSUE THREAD

**Issue by vincent-ogury** [nestjs/mapped-types#1124]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

```
import { IntersectionType, PartialType } from '@nestjs/mapped-types';
import { IsOptional, IsString } from 'class-validator';

class MandatoryField {
  @IsString()
  public my_test: string;
}

class OtherFields {
  @IsOptional()
  @IsString()
  public my_optional_field?: string;
}

class MyCombinaison extends IntersectionType(
  PartialType(MandatoryField),
  OtherFields,
) {}

function t(r: MyCombinaison) {
  console.debug(r);
}

t({});
# Argument of type '{}' is not assignable to parameter of type 'MyCombinaison'.
  Type '{}' is missing the following properties from type 'MyCombinaison': my_test, my_optional_field
```

Here I expect to have only optional parameter, but they are all mandatory.
When I use @nestjs/swagger, no issue.
Also version < 2 works as expected.

### Minimum reproduction code

https://see.example.above

### Steps to reproduce

1. install last version of nestjs
2. install last version of @nestjs/mapped-types

### Expected behavior

All fields above should be optional.

### Package version

2.0.1

### Node.js version

v18.16.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by Keeper991** [comment]
I faced the same problem. There is a problem in that the properties of the instance of the class inherited as a partial type are not guided to an option.

**Comment by micalevisk** [comment]
@Keeper991 do you mind on sharing a minimum reproduction repository? it would help us to fix that quickly

**Comment by celogeek** [comment]
@micalevisk did the example above help ? it show the problem immediately.

**Comment by micalevisk** [comment]
@vincent-ogury 
@celogeek 

# minimum reproduction **repository**

(note the _repository_ part_)

please don't spam. That doesn't help on fixing the issue. Also, you guys could investigate why that bug exists and try to fix that as well :)

**Comment by Keeper991** [comment]
https://github.com/Keeper991/partial-type-issue/blob/master/src/app.service.ts

ChildClass inherits ParentClass with PartialType, and tried to push partial object in ChildClass Array.
but ts throws an error message 'missing the following properties'.

**Comment by micalevisk** [comment]
thanks!

that was a regression introduced by PR #1027 on v2.0.0

**Comment by micalevisk** [comment]
and that's due to your `strictNullChecks` being `false` btw

**Comment by kamilmysliwiec** [comment]
Fixed in 2.0.2
